internal/xcrd/crd.go

```diff
@@ -233,6 +233,25 @@ func getProps(field string, v *v1.CompositeResourceValidation) (map[string]extv1
 	return spec.Properties, spec.Required, nil
 }
 
+// setCrdMetadata sets the labels and annotations on the CRD.
+func setCrdMetadata(crd *extv1.CustomResourceDefinition, xrd *v1.CompositeResourceDefinition) *extv1.CustomResourceDefinition {
+	crd.SetLabels(xrd.GetLabels())
+	if xrd.Spec.Metadata != nil {
+		if xrd.Spec.Metadata.Labels != nil {
+			if crd.GetLabels() == nil {
+				crd.SetLabels(map[string]string{})
+			}
+			for k, v := range xrd.Spec.Metadata.Labels {
+				crd.GetLabels()[k] = v
+			}
```

Comment:
Minor nit: What  about something like this instead?
```suggestion
			inheritedLabels := crd.GetLabels()
			if  inheritedLabels == nil {
				inheritedLabels = map[string]string{}
			}
			for k, v := range xrd.Spec.Metadata.Labels {
				inheritedLabels[k] = v
			}
			crd.SetLabels(inheritedLabels)
```

Reply:
I don't have a strong opinion about it :) Sure why no, it maybe improves readability.